### PR TITLE
Create post-registration flow and profile update form (same thing)

### LINKED ISSUE #174: Create login and registration flows
# Feature
## What should your feature do?
- Create routes at `/login`, `/sign_up`, and `/join`.
- Create a modal-focused login flow
- Redesign and simplify current registration flow
- Add section for [data use info](https://github.com/OperationCode/operationcode_frontend/issues/983)
- Collaborate with @Wimsy113 and @lauraux to ensure design/product vision is being executed properly.
- Be aware of the discussion [here](https://github.com/OperationCode/operationcode_frontend/issues/877) and [here](https://github.com/OperationCode/operationcode_frontend/issues/867) and [here](https://github.com/OperationCode/operationcode_frontend/issues/754) and [here](https://github.com/OperationCode/operationcode_frontend/issues/410) and [here](https://github.com/OperationCode/operationcode_frontend/issues/409)

### PR DESCRIPTION
**Pull request by kylemh**
# Read First
I've chunked up a lot of this PR in #503 #505 and #506 
Please review those first. This PR will incorporate them and decrease the number of files changed.

# Description Of Changes
Create a `/profile/update` route which can be used when logged in to change a user's info. This view is navigated to after registering (to encourage a user to provide more info about themselves).

# Issue Resolved

Fixes #174

### CODE REVIEW

**Comment by kylemh** [comment]
🤖 **This is a bot** 🤖<br /><br />🎉 Deployed Storybook preview! 🎉<br /><br />Click the link at the bottom of this comment to see it live.

Built with commit 761706b2f5be4a87569ddec2b4b615c031488631

https://deploy-preview-499--operation-code-storybook.netlify.com

**Comment by kylemh** [comment]
🤖 **This is a bot** 🤖<br /><br />🎉 Deployed Storybook preview! 🎉<br /><br />Click the link at the bottom of this comment to see it live.

Built with commit 731a833f9c775527c5ef6013fb5eb92f17c5f6df

https://deploy-preview-499--operation-code-storybook.netlify.com

**Comment by kylemh** [comment]
🤖 **This is a bot** 🤖<br /><br />🎉 Deployed Storybook preview! 🎉<br /><br />Click the link at the bottom of this comment to see it live.

Built with commit 5e302ced0201fd34fa731e642961dcee869fcf7f

https://deploy-preview-499--operation-code-storybook.netlify.com

File: components/UpdateProfileForm/steps/MilitaryDetails.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        <h2 className={styles.row}>Military Details</h2>
+
+        <div className={styles.row}>
+          <Field
+            className={styles.fullWidth}
+            name="branchOfService"
+            label="Branch Of Service*"
+            component={Select}
+            options={[
+              {
```
Comment (AllenAnthes) [inline components/UpdateProfileForm/steps/MilitaryDetails.js]: Is there a reason this is declared as an option instead of a `placeholder`?

Comment (kylemh) [reply components/UpdateProfileForm/steps/MilitaryDetails.js]: Good shout! I can probably just remove this option and leave the default placeholder

**Comment by kylemh** [comment]
Will make more tests in separate PR

**Pull request merged**